libs/rtype/sources/Systems/AttackSystem.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+** EPITECH PROJECT, 2023
+** engine
+** File description:
+** AttackSystem
+*/
+
+#include "Systems/AttackSystem.hpp"
+
+#include "Components/Collision.hpp"
+#include "Components/HitBox.hpp"
+#include "Components/Position.hpp"
+#include "Components/Velocity.hpp"
+#include "Components/Drawable.hpp"
+#include "Components/Controllable.hpp"
+
+System::AttackSystem::AttackSystem(
+    SparseArray<Component::Attack> &attacks, engine::Registry &reg
+)
+    : _attacks(attacks)
+    , _register(reg)
+{
+}
+
+void System::AttackSystem::operator()()
+{
+    for (auto it = _attacks.begin(); it != _attacks.end(); it++) {
+        if (it->value().is_attacking) {
+            createLaserEntity(it->value(), engine::Entity(it.get_entity()));
+        }
+    }
+}
+
+void System::AttackSystem::createLaserEntity(
+    Component::Attack &attack_comp, engine::Entity const attacker_index
+)
+{
+    size_t laser_index = 2;
+    Component::Velocity velocity(1, 0);
+    SparseArray<Component::Position> &positions
+        = _register.get_components<Component::Position>();
+    engine::Entity attack_entity(_register.get_new_entity());
+    Component::Position &attacker_pos = positions[attacker_index].value();
+    std::optional<Component::Collision> attacker_collision =
+        _register.get_components<Component::Collision>()[attacker_index];
+    Component::Position attack_entity_pos(attacker_pos._x, attacker_pos._y);
```

Comment:
tu peux juste utiliser le copy ctor du coup
```suggestion
    Component::Position attack_entity_pos(attacker_pos);
```

Reply:
réel